main/config.go

```diff
@@ -230,7 +243,38 @@ func (c *Config) setDefaultURLs() error {
 	return nil
 }
 
-// loadAuthMap loads the auth map from the environment
+// loadDeviceIdentitiesFile loads device identities from the identities JSON file.
+// Returns without error if file does not exist.
+func (c *Config) loadDeviceIdentitiesFile(configDir string) error {
+	// if file does not exist, return right away
+	if _, err := os.Stat(identitiesFile); os.IsNotExist(err) {
+		return nil
+	}
+
+	identitiesBytes, err := ioutil.ReadFile(filepath.Join(configDir, identitiesFile))
+	if err != nil {
+		return err
+	}
+
+	var buffer []map[string]string
+	err = json.Unmarshal(identitiesBytes, &buffer)
+	if err != nil {
+		return err
+	}
```

Comment:
Since File is a writer, you can directly unmarshal the file without copying the complete content into memory:

```go
fileHandle, err = os.Open(filepath.Join(configDir, identitiesFile))
if err != nil {
    return err
}
defer fileHandle.close()

jd = json.NewDecoder(fileHandle)
err = jd.Decode(&buffer)

// or more concise:
json.NewDecoder(fileHandle).Decode(&buffer)
```

Reply:
..and can we rename `buffer` info `keyValue` or something else? I first confused it with a binary buffer

Reply:
thanks! take a look now please